components/post_view/post/post.jsx

```diff
@@ -256,8 +257,13 @@ export default class Post extends React.PureComponent {
         this.setState({a11yActive: false});
     }
 
+    handlePostFocus = () => {
+        const {post, author, reactions, isFlagged} = this.props;
+        this.setState({currentAriaLabel: PostUtils.createAriaLabelForPost(post, author, isFlagged, reactions, this.context.intl)});
```

Comment:
no need for intl and you don't need to set this in the state.. 

you can do
arial-label=PostUtils.createAriaLabelForPost(post, author, reactions, isFlagged) in the component prop in the render method

Reply:
i'm doing it as an `onFocus()` method so that it doesn't load them all at once. this addresses https://github.com/mattermost/mattermost-webapp/pull/3090#discussion_r304611979

Reply:
ohh ok then it makes total sense

Reply:
I still need intl for date/time formatting